protocols/identify/src/identify.rs

```diff
@@ -254,6 +272,16 @@ impl NetworkBehaviour for Identify {
     ) {
         match event {
             IdentifyHandlerEvent::Identified(info) => {
+                match self.discovered_peers.get_mut(&peer_id) {
+                    Some(entry) => {
+                        entry.extend(info.listen_addrs.clone());
```

Comment:
Do I understand correctly that the `HashSet` of `Multiaddr`s for a given peer could potentially grow endlessly, e.g. when the remote (malicious) peer sends a new set of addresses on each `identify` exchange? How about replacing the previous set with the new one instead? Would as well make sure outdated addresses are removed.

Reply:
Replacing the set is a good idea!